File: testdata/vulnerable.php
<?php

function bad_md5($data) {
    return md5($data);
}

function bad_sha1($data) {
    return sha1($data);
}

function bad_md5_hash($data) {
    return hash('md5', $data);
}

function bad_des($plaintext, $key) {
    return openssl_encrypt($plaintext, 'des-cbc', $key);
}

function bad_3des($plaintext, $key) {
    return openssl_encrypt($plaintext, 'des-ede3-cbc', $key);
}

function bad_rc4($plaintext, $key) {
    return openssl_encrypt($plaintext, 'rc4', $key);
}

function legacy_mcrypt($plaintext, $key) {
    $iv = mcrypt_create_iv(8, MCRYPT_DES);
    return mcrypt_encrypt(MCRYPT_DES, $key, $plaintext, MCRYPT_MODE_CBC, $iv);
}

File: testdata/vulnerable.rb
require 'digest/md5'
require 'openssl'

def bad_md5(data)
  Digest::MD5.hexdigest(data)
end

def bad_sha1(data)
  Digest::SHA1.hexdigest(data)
end

def bad_des(key, iv)
  OpenSSL::Cipher.new('DES-CBC').encrypt
end

def bad_3des(key, iv)
  OpenSSL::Cipher.new('DES-EDE3-CBC').encrypt
end

def bad_rc4(key)
  OpenSSL::Cipher.new('RC4').encrypt
end

def weak_rsa
  OpenSSL::PKey::RSA.new(1024)
end

def disallowed_dsa
  OpenSSL::PKey::DSA.new(2048)
end

def non_nist_curve
  OpenSSL::PKey::EC.new('secp256k1')
end
